python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -370,8 +372,60 @@ def parse_node_data(in_file, feat_ops, node_id_col, label_conf,
             feat_data[key] = val
     return (data[node_id_col], feat_data)
 
+def map_node_ids(src_ids, dst_ids, edge_type, node_id_map, skip_nonexist_edges):
+    """ Map node IDs of source and destination nodes of edges.
+
+    In the ID mapping, we need to handle many errors in the input data. For example,
+    we need to handle the case that endpoint nodes of edges don't exist; we also need
+    to handle the case that the data type of node IDs of the endpoint nodes don't
+    match the data type of the keys of the ID map.
+
+    Parameters
+    ----------
+    src_ids : tensor
+        The source nodes.
+    dst_ids : tensor
+        The destination nodes.
+    edge_type : tuple
+        It contains source node type, relation type, destination node type.
+    node_id_map : dict
+        The key is the node type and value is IdMap or IdentityMap.
+    skip_nonexist_edges : bool
+        Whether or not to skip edges whose endpoint nodes don't exist.
+
+    Returns
+    -------
+    tuple of tensors : the remapped source and destination node IDs.
+    """
+    src_type, _, dst_type = edge_type
+    new_src_ids, idx = node_id_map[src_type](src_ids)
+    # If some of the source nodes don't exist in the node set.
+    if len(idx) != len(src_ids):
+        bool_mask = np.ones(len(src_ids), dtype=bool)
+        bool_mask[idx] = False
+        if skip_nonexist_edges:
+            print(f"source nodes of {src_type} do not exist: {src_ids[bool_mask]}")
+        else:
+            raise ValueError(f"source nodes of {src_type} do not exist: {src_ids[bool_mask]}")
+        dst_ids = dst_ids[idx]
+    src_ids = new_src_ids
+
+    new_dst_ids, idx = node_id_map[dst_type](dst_ids)
+    # If some of the dest nodes don't exist in the node set.
+    if len(idx) != len(dst_ids):
+        bool_mask = np.ones(len(dst_ids), dtype=bool)
+        bool_mask[idx] = False
+        if skip_nonexist_edges:
+            print(f"dest nodes of {src_type} do not exist: {dst_ids[bool_mask]}")
+        else:
+            raise ValueError(f"dest nodes of {src_type} do not exist: {dst_ids[bool_mask]}")
+        # We need to remove the source nodes as well.
+        src_ids = src_ids[idx]
+    dst_ids = new_dst_ids
+    return src_ids, dst_ids
+
 def parse_edge_data(in_file, feat_ops, src_id_col, dst_id_col, edge_type,
```

Comment:
This function now has 9 arguments. Refactoring this will be difficult. Consider creating a wrapper configuration object that groups together relevant arguments (using a `dataclass`) or breaking into multiple functions.

A good rule of thumb (that [pylint](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/too-many-arguments.html) uses) is for function to not have more than 5 arguments if possible.

A good way to avoid having to pass state around like this is to wrap functions like these in a class that you can initialize with all the shared state.

Reply:
i try to understand the problem of this. is it about the number of arguments or the amount of work done in this function?
if we wrap multiple arguments into an object, it doesn't change anything except the number of arguments.

we cannot break this function because this is a entry function that runs in process pool.

Reply:
i guess the easiest solution is to pass `process_conf` to the function directly.